### PR TITLE
Split dynamic import into its own page

### LINKED ISSUE #10550: Issue with "import": import is not a statement but a declaration
MDN URL: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/import

The MDN documentation currently refers to `import` as a statement that can also be used like a function. I believe this wording is responsible for quite a bit of confusion, since a static import is never a statement but an **import declaration**, to use the vocabulary of the ECMAScript standard. Import and export declarations are only allowed at the base of modules and not in any other place a statement would be allowed, and imports are subject to hoisting.

On the other hand, **import calls** are allowed anywhere an expression is allowed, even outside of modules, and their syntax works like any function call. They can be executed conditionally, their path can be specified by a template literal, and they return a promise.

I'm not sure if separating the article in two is necessary, but I will point out that MDN does have separate pages for function declarations and function expressions, and I remember when I was learning JS years ago that it really made it clear to me that those were two parts of language that just happened to use the same word. Food for thoughts.

<details>
<summary>MDN Content page report details</summary>

* Folder: `en-us/web/javascript/reference/statements/import`
* MDN URL: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/import
* GitHub URL: https://github.com/mdn/content/blob/main/files/en-us/web/javascript/reference/statements/import/index.md
* Last commit: https://github.com/mdn/content/commit/5344089592af3da25052ead184d73b5e45eebd9f
* Document last modified: 2021-08-08T09:02:39.000Z

</details>

### PR DESCRIPTION
**Pull request by Josh-Cena**
#### Motivation

Fix #10550

Dynamic import and static imports are very different: the former is an expression, the latter is a declaration. Dynamic imports are allowed in non-modules. Dynamic imports allow arbitrary expressions as specifiers. All these differences make it reasonable to split them into two pages.

#### Metadata

- [x] Adds a new document
- [ ] Rewrites (or significantly expands) a document
- [ ] Fixes a typo, bug, or other error

### CODE REVIEW

File: files/en-us/web/javascript/reference/operators/import/index.md [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+
+The `import()` call is a function-like expression that allows asynchronously loading an ECMAScript module into a potentially non-module environment.
+
+## Syntax
+
+```js
+import(moduleName)
+```
+
+The `import()` call is a syntax that closely resembles a function call, but `import` itself is a keyword, not a function. You cannot alias it like `const myImport = import`, which will throw a `SyntaxError`.
```
Comment (teoli2003) [inline files/en-us/web/javascript/reference/operators/import/index.md:24]:
```suggestion
The `import()` call is a syntax that closely resembles a function call, but `import` itself is a keyword, not a function. You cannot alias it like `const myImport = import`, which will throw a {{jsxref("SyntaxError")}}.
```

File: files/en-us/web/javascript/reference/operators/import/index.md [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+slug: Web/JavaScript/Reference/Operators/import
+tags:
+  - ECMAScript 2015
+  - JavaScript
+  - Language feature
+  - Modules
+  - Reference
+  - dynamic import
+  - import
+browser-compat: javascript.operators.import
```
Comment (Josh-Cena) [inline files/en-us/web/javascript/reference/operators/import/index.md:12]: I suppose this also needs work from BCD to split the compat table?

File: files/en-us/web/javascript/reference/operators/import/index.md [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+---
+{{jsSidebar("Operators")}}
+
+The `import()` call is a function-like expression that allows asynchronously loading an ECMAScript module into a potentially non-module environment.
+
```
Comment (teoli2003) [inline files/en-us/web/javascript/reference/operators/import/index.md:17]: You should add some sentences to explain that this is the _dynamic import_ (vs the _static import_) as you use the distinction later in the page.

File: files/en-us/web/javascript/reference/statements/import/index.md [hunk 4]
```diff
@@ -14,18 +14,15 @@ browser-compat: javascript.statements.import
...
 
 Imported modules are in {{JSxRef("Strict_mode","strict mode")}}
 whether you declare them as such or not. The `import` statement cannot be
 used in embedded scripts unless such script has a `type="module"`. Bindings
 imported are called live bindings because they are updated by the module that exported
 the binding.
 
-There is also a function-like dynamic **`import()`**, which
-does not require scripts of `type="module"`.
+There is also a function-like dynamic [**`import()`**](/en-US/docs/Web/JavaScript/Reference/Operators/import), which does not require scripts of `type="module"`.
```
Comment (teoli2003) [inline files/en-us/web/javascript/reference/statements/import/index.md:25]: No need for bold here.
```suggestion
There is also a function-like dynamic [`import()`](/en-US/docs/Web/JavaScript/Reference/Operators/import), which does not require scripts of `type="module"`.
```

**Review by teoli2003 — CHANGES_REQUESTED** [review]
Yes, it is a good idea. I've added a few comments and suggestions.

File: files/en-us/web/javascript/reference/operators/import/index.md (hunk 2 above)
Comment (teoli2003) [reply files/en-us/web/javascript/reference/operators/import/index.md:12]: Yes, we need a new file there: https://github.com/mdn/browser-compat-data/tree/main/javascript/operators

**Comment by teoli2003** [comment]
Looks good to me. Let's wait until the PR for bcd is created before formally reviewing and merging this.

File: files/en-us/web/javascript/reference/operators/import/index.md (hunk 2 above)
Comment (Josh-Cena) [reply files/en-us/web/javascript/reference/operators/import/index.md:12]: https://github.com/mdn/browser-compat-data/pull/16720

**Comment by Josh-Cena** [comment]
I also noticed `import.meta` is in "statements & declarations" instead of "expressions & operators". I will follow-up since that one requires BCD refactor as well.

**Comment by queengooborg** [comment]
BCD PR has now been merged!

**Comment by teoli2003** [comment]
Thank you

**Review by teoli2003 — APPROVED** [review]

**Pull request merged**